### PR TITLE
Update fork for JBoss builds

### PR DESCRIPTION
**Pull request by jasondlee**

### CODE REVIEW

File: cdi/pom.xml [hunk 1]
```diff
@@ -16,16 +16,16 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
     <parent>
-        <groupId>org.glassfish</groupId>
+        <groupId>org.wildfly.faces</groupId>
         <artifactId>mojarra-parent</artifactId>
-        <version>4.0.1-SNAPSHOT</version>
+        <version>4.0.0.SP01-SNAPSHOT</version>
```
Comment (fjuma) [inline cdi/pom.xml:26]: Just to check, this was based off 4.0.0-M6, right? I'm guessing there was just an extra commit in their branch that bumped to 4.0.1?

File: impl/pom.xml [hunk 2]
```diff
@@ -27,17 +27,23 @@
    <modelVersion>4.0.0</modelVersion>
 
    <parent>
-        <groupId>org.glassfish</groupId>
+        <groupId>org.wildfly.faces</groupId>
         <artifactId>mojarra-parent</artifactId>
-        <version>4.0.1-SNAPSHOT</version>
-    </parent>
+        <version>4.0.0.SP01-SNAPSHOT</version>
```
Comment (fjuma) [inline impl/pom.xml:32]: Just to check, this was based off 4.0.0-M6, right? I'm guessing there was just an extra commit in their branch that bumped to 4.0.1?

File: impl/pom.xml [hunk 3]
```diff
@@ -27,17 +27,23 @@
    <modelVersion>4.0.0</modelVersion>
 
    <parent>
-        <groupId>org.glassfish</groupId>
+        <groupId>org.wildfly.faces</groupId>
```
Comment (fjuma) [inline impl/pom.xml]: I think we can stick to `com.sun.faces`.

**Comment by jasondlee** [comment]
I think that's fight, @fjuma.  I have a local rebase off of upstream/master that makes quite a few changes. Not sure how best to handle that.

**Comment by fjuma** [comment]
> I think that's fight, @fjuma. I have a local rebase off of upstream/master that makes quite a few changes. Not sure how best to handle that.

Since this branch is indeed based off 4.0.0-M6 and the 4.0.1-SNAPSHOT was just part of an extra commit where they bumped the version in the M6 branch, the 4.0.0.SP01-SNAPSHOT version here is fine. No need to incorporate the extra commits from master yet.

File: impl/pom.xml [hunk 4]
```diff
@@ -27,17 +27,23 @@
...
    <parent>
-        <groupId>org.glassfish</groupId>
+        <groupId>com.sun.faces</groupId>
         <artifactId>mojarra-parent</artifactId>
-        <version>4.0.1-SNAPSHOT</version>
-    </parent>
+        <version>4.0.0.SP01-SNAPSHOT</version>
+   </parent>
 
-   <artifactId>jakarta.faces</artifactId>
```
Comment (fjuma) [inline impl/pom.xml:35]: I think `<groupId>com.sun.faces</groupId>` should be added here too.

**Comment by jasondlee** [comment]
> Since this branch is indeed based off 4.0.0-M6 and the 4.0.1-SNAPSHOT was just part of an extra commit where they bumped the version in the M6 branch, the 4.0.0.SP01-SNAPSHOT version here is fine. No need to incorporate the extra commits from master yet.

I compared the repo between the hash for the tag and the head of master and got mostly bug fixes, etc. I think, then, we can merge this PR to get the repo ready for work, then open another to get it up-to-date with master?

File: impl/pom.xml [hunk 5]
```diff
@@ -400,8 +420,12 @@
                         <manifestFile>${project.build.outputDirectory}/META-INF/MANIFEST.MF</manifestFile>
                     </archive>
                     <excludes>
-                        <exclude>**/*.java</exclude>
-                        <exclude>**/*.html</exclude>
+                        <exclude>jakarta/**</exclude>
+                        <exclude>com/sun/faces/vendor/GlassFish*</exclude>
+                        <exclude>com/sun/faces/vendor/Jetty*</exclude>
+                        <exclude>com/sun/faces/vendor/Tomcat*</exclude>
```
Comment (fjuma) [inline impl/pom.xml]: Might be good to check if the above three files still exist in the `impl/src/main/java/com/sun/faces/vendor/` directory.

File: impl/pom.xml (hunk 4 above)
Comment (jasondlee) [reply impl/pom.xml:35]: I can, but tf the groupId is the same as the parent, specifying it again is not necessary.

File: impl/pom.xml (hunk 3 above)
Comment (jasondlee) [reply impl/pom.xml]: Fixed

File: impl/pom.xml (hunk 5 above)
Comment (jasondlee) [reply impl/pom.xml]: They do not. I'll remove the exclusions.

File: impl/pom.xml [hunk 6]
```diff
@@ -411,6 +435,16 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-source-plugin</artifactId>
                 <version>3.2.1</version>
+                <configuration>
+                   <excludes>
+                      <exclude>jakarta/**</exclude>
+                      <exclude>com/sun/faces/vendor/GlassFish*</exclude>
+                      <exclude>com/sun/faces/vendor/Jetty*</exclude>
+                      <exclude>com/sun/faces/vendor/Tomcat*</exclude>
```
Comment (fjuma) [inline impl/pom.xml]: It might be possible to remove the above three excludes here too if these files no longer exist.

File: pom.xml [hunk 7]
```diff
@@ -18,22 +18,25 @@
...
+    <modelVersion>4.0.0</modelVersion>
 
-   <parent>
-        <groupId>org.eclipse.ee4j</groupId>
-        <artifactId>project</artifactId>
-        <version>1.0.6</version>
+    <parent>
+        <groupId>org.jboss</groupId>
+        <artifactId>jboss-parent</artifactId>
+        <version>22</version>
```
Comment (fjuma) [inline pom.xml]: We can update the version here. I think we can update to 39?

File: pom.xml [hunk 8]
```diff
@@ -18,22 +18,25 @@
...
     </parent>
 
-   <groupId>org.glassfish</groupId>
-   <artifactId>mojarra-parent</artifactId>
-   <version>4.0.1-SNAPSHOT</version>
-   <packaging>pom</packaging>
+    <groupId>com.sun.faces</groupId>
+    <artifactId>mojarra-parent</artifactId>
```
Comment (fjuma) [inline pom.xml]: I think this should be `jsf-project`.

File: impl/pom.xml [hunk 9]
```diff
@@ -27,17 +27,23 @@
    <modelVersion>4.0.0</modelVersion>
 
    <parent>
-        <groupId>org.glassfish</groupId>
+        <groupId>com.sun.faces</groupId>
         <artifactId>mojarra-parent</artifactId>
```
Comment (fjuma) [inline impl/pom.xml]: I think this should be `jsf-project`.

File: pom.xml [hunk 10]
```diff
@@ -45,22 +48,140 @@
             <distribution>repo</distribution>
         </license>
     </licenses>
+    <repositories>
+        <repository>
+            <id>sonatype-staging</id>
```
Comment (fjuma) [inline pom.xml:53]: Is this one needed?

File: impl/pom.xml (hunk 9 above)
Comment (jasondlee) [reply impl/pom.xml]: Fixed.

File: pom.xml (hunk 10 above)
Comment (jasondlee) [reply pom.xml:53]: For now, yes, it seems to be necessary to pull in the EE 10 artifacts. Once things settle down in that area, this can be removed.

File: pom.xml (hunk 8 above)
Comment (jasondlee) [reply pom.xml]: Fixed

File: pom.xml (hunk 7 above)
Comment (jasondlee) [reply pom.xml]: Fixed

File: impl/pom.xml (hunk 6 above)
Comment (jasondlee) [reply impl/pom.xml]: Fixed

File: impl/pom.xml (hunk 4 above)
Comment (jasondlee) [reply impl/pom.xml:35]: Fixed

File: pom.xml [hunk 11]
```diff
@@ -45,22 +48,140 @@
...
         <maven.compiler.target>11</maven.compiler.target>
     </properties>
 
-   <build>
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-enforcer-plugin</artifactId>
```
Comment (fjuma) [inline pom.xml]: I know we had this in our 2.3.x forks but I don't think we had it in the 3.0.x forks. We should be able to remove this.

File: pom.xml (hunk 10 above)
Comment (fjuma) [reply pom.xml:53]: Got it, makes sense.

**Comment by fjuma** [comment]
> > Since this branch is indeed based off 4.0.0-M6 and the 4.0.1-SNAPSHOT was just part of an extra commit where they bumped the version in the M6 branch, the 4.0.0.SP01-SNAPSHOT version here is fine. No need to incorporate the extra commits from master yet.
> 
> I compared the repo between the hash for the tag and the head of master and got mostly bug fixes, etc. I think, then, we can merge this PR to get the repo ready for work, then open another to get it up-to-date with master?

Yeah, if it will be helpful to incorporate the latest commits that haven't been included in a tag yet, then you could open a follow up PR to incorporate them.

**Comment by fjuma** [comment]
@jasondlee Just to check, have you had a chance to take a look at our 3.0.0.SP04 fork to see if the extra commits that we had included there have now been merged upstream? Just wondering if there are still extra fixes that we should cherry-pick here.

**Comment by fjuma** [comment]
Another thing that would be good to check if you haven't already done so is the impl JAR to make sure that it really does only contain the impl classes.

**Comment by jasondlee** [comment]
> @jasondlee Just to check, have you had a chance to take a look at our 3.0.0.SP04 fork to see if the extra commits that we had included there have now been merged upstream? Just wondering if there are still extra fixes that we should cherry-pick here.

I think I have, but I'll double check to be sure.

File: pom.xml (hunk 11 above)
Comment (jasondlee) [reply pom.xml]: Removed.

**Comment by jasondlee** [comment]
> Another thing that would be good to check if you haven't already done so is the impl JAR to make sure that it really does only contain the impl classes.

Verified.

**Comment by jasondlee** [comment]
> > @jasondlee Just to check, have you had a chance to take a look at our 3.0.0.SP04 fork to see if the extra commits that we had included there have now been merged upstream? Just wondering if there are still extra fixes that we should cherry-pick here.
> 
> I think I have, but I'll double check to be sure.

I think we're good on the cherry picks.

File: impl/src/main/java/com/sun/faces/context/flash/ELFlash.java [hunk 12]
```diff
@@ -1289,7 +1289,7 @@ void expireNext_MovePreviousToNext() {
                 flashMap.clear();
                 // remove it from the flash
                 innerMap.remove(nextRequestFlashInfo.getSequenceNumber() + "");
-                nextRequestFlashInfo = null;
+                expireNext();
```
Comment (fjuma) [inline impl/src/main/java/com/sun/faces/context/flash/ELFlash.java:1292]: Is this something that was previously in our fork that we're cherry-picking here? If so, we should keep that as a separate commit so it stands out as an additional commit in our branch.

Comment (jasondlee) [reply impl/src/main/java/com/sun/faces/context/flash/ELFlash.java:1292]: That is a cherry-pick, yes.

**Review by fjuma — APPROVED** [review]
